### ISSUE TITLE
`ntpath.join()` adds slash to incomplete UNC drive with trailing slash

### ISSUE THREAD

**Issue by barneygale** [python/cpython#103220]
`ntpath.join()` incorrectly inserts an additional slash when joining an argument onto an incomplete UNC drive with a trailing slash:

```python
>>> import ntpath
>>> ntpath.join('\\\\server\\share\\foo\\', 'bar')
'\\\\server\\share\\foo\\bar'  # ok
>>> ntpath.join('\\\\server\\share\\', 'foo')
'\\\\server\\share\\foo'  # ok
>>> ntpath.join('\\\\server\\', 'share')
'\\\\server\\\\share'  # wrong!
>>> ntpath.join('\\\\', 'server')
'\\\\\\server'  # wrong!
```

Before 005e69403d638f9ff8f71e59960c600016e101a4 (3.12), the last test case succeeds because `splitdrive()` doesn't identify `'\\\\'` as a UNC drive. But the third test case is reproducible going back to 3.11 and 3.10.

### Linked PRs
* gh-103221

**Comment by eryksun** [comment]
This would be an enhancement to add the ability to reliably join segments as a UNC drive. I don't think it has to be backported to 3.11 and 3.10.

**Comment by barneygale** [comment]
Resolved in [b57105a](https://github.com/python/cpython/commit/b57105ae33e1f61e6bdf0eec45c4135d067b9b22) / #103221.
